AppController/lib/haproxy.rb

```diff
@@ -321,4 +347,121 @@ def self.initialize_config()
     # Write the base configuration file which sets default configuration parameters
     File.open(BASE_CONFIG_FILE, "w+") { |dest_file| dest_file.write(base_config) }
   end
+
+
+  # Retrieves HAProxy stats for the given app.
+  #
+  # Args:
+  #   app_name: The name of the app to get HAProxy stats for.
+  # Returns:
+  #   The total requests for the app, the requests enqueued and the
+  #    timestamp of stat collection.
+  def self.get_haproxy_stats(app_name)
+    Djinn.log_debug("Getting scaling info for application #{app_name}")
+
+    total_requests_seen = 0
+    total_req_in_queue = 0
+    time_requests_were_seen = 0
+
+    # Retrieve total and enqueued requests for the given app.
+    monitoring_info = Djinn.log_run("echo \"show stat\" | " +
+      "socat stdio unix-connect:/etc/haproxy/stats | grep #{app_name}")
+    Djinn.log_debug("HAProxy raw stats: #{monitoring_info}")
+
+    if monitoring_info.empty?
+      Djinn.log_warn("Didn't see any monitoring info - #{app_name} may not " +
+        "be running.")
+      return :no_change, :no_change, :no_backend
+    end
+
+    monitoring_info.each_line { |line|
+      parsed_info = line.split(',')
+      # no request info here
```

Comment:
It's an old comment: we used to also ask for 'show info' which adds data where no fitting in the 50 colums (https://cbonte.github.io/haproxy-dconv/configuration-1.4.html#9.2) returns. The check prevent us to try to parse them. I don't think the comment and the check is still needed, but it doesn't harm to leave it (just in case).